[feature branch] Upgrade chakra to v3

alexvuong opened the pull request:
<!--- Provide a short summary of your changes in the Title field above -->

# Description

<!--- A longer summary of your changes, including: a description of the issue that you’re addressing, a list of required dependencies (if applicable), and any other relevant context. -->

# Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] **Bug fix** (non-breaking change that fixes an issue)
- [ ] **New feature** (non-breaking change that adds functionality)
- [ ] **Documentation update**
- [ ] **Breaking change** (could cause existing functionality to not work as expected)
- [ ] **Other changes** (non-breaking changes that does not fit any of the above)

> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- (step1)

# Checklists

<!--- Enter an `x` in all the boxes that apply. -->
<!--- If you’re unsure about any of these, don’t hesitate to ask. We’re here to help! -->

## General

- [ ] Changes are covered by test cases
- [ ] CHANGELOG.md updated with a short description of changes (_not_ required for documentation updates)

## Accessibility Compliance

You must check off all items in **one** of the follow two lists:

- [ ] There are no changes to UI

_or..._

- [ ] Changes were tested with a Screen Reader (iOS VoiceOver or Android Talkback) and had no issues
- [ ] Changes comply with [WCAG 2.0 guidelines levels A and AA](https://www.wuhcag.com/wcag-checklist/)
- [ ] Changes to common UI patterns and interactions comply with [WAI-ARIA best practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## Localization

- [ ] Changes include a UI text update in the Retail React App (which requires translation)

cc-prodsec commented:
![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/pr-banner-default.svg)

### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/cc-mobify-sdk/pr-checks/6a4b22cc-68e1-4250-aa99-697b4618be56?refs=4e16c4a9f7e60e2f392311d43a76d453126a07ff&source=prChecksComment)

:white_check_mark: **license/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/cc-mobify-sdk/pr-checks/6a4b22cc-68e1-4250-aa99-697b4618be56/license?refs=4e16c4a9f7e60e2f392311d43a76d453126a07ff&source=prChecksComment)

alexvuong commented:
NOTE: The CI will be broken because we are commenting out a lot of code, which will be fixed in another tickets

alexvuong commented on packages/template-typescript-minimal/package.json:
```diff
@@ -19,25 +19,54 @@
     "@types/react": "~18.2.0",
     "@types/react-dom": "~18.2.1",
     "@types/react-router-dom": "~5.3.3",
+    "@peculiar/webcrypto": "^1.4.2",
```
TODO: Keep this set up until we are ready to merge this upgrade to next branch

bendvc committed:
Fix lint

bendvc committed:
Initial Commit

kevinxh committed:
update toast action

kevinxh committed:
update comment

kevinxh committed:
remove home page toast example

kevinxh committed:
Merge pull request #2489 from SalesforceCommerceCloud/feature/chakra-ui-upgrade-v3-toasts

@W-18602313 Chakra V3 Upgrade - toast

bendvc committed:
Merge branch 'feature/chakra-ui-upgrade-v3' into bendvc/W-18672066_image_gallery_upgrade

bendvc committed:
Merge branch 'feature/chakra-ui-upgrade-v3' into bendvc/W-18672066_quantity_picker_upgrade

kevinxh committed:
enable checkout page

kevinxh committed:
uncomment the checkout header

bendvc committed:
Initial commit

bendvc committed:
Update packages/extension-chakra-storefront/src/components/image-gallery/index.jsx

Co-authored-by: Adam Raya <adamraya@users.noreply.github.com>
Signed-off-by: Ben Chypak <bchypak@mobify.com>

bendvc committed:
Remove the example page.

bendvc committed:
Merge branch 'bendvc/W-18672066_image_gallery_upgrade' of https://github.com/SalesforceCommerceCloud/pwa-kit into bendvc/W-18672066_image_gallery_upgrade

bendvc committed:
Added work around for active style, remove sample code

bendvc committed:
Merge pull request #2498 from SalesforceCommerceCloud/bendvc/W-18672066_image_gallery_upgrade

[Chakra V3 Upgrade 🎨] Update `ImageGallery` Component @@W-18672066@@

bendvc committed:
Merge branch 'feature/chakra-ui-upgrade-v3' into bendvc/W-18672066_quantity_picker_upgrade

bendvc committed:
Merge pull request #2504 from SalesforceCommerceCloud/bendvc/W-18672066_quantity_picker_upgrade

[Chakra V3 Upgrade 🎨] Update QuantityPicker Component @@W-18682956@@

bendvc committed:
Re add context, update sample

bendvc committed:
Fix expanding/collapsing

adamraya committed:
[Chakra V3 Upgrade 🎨] Update Breadcrumb component (@W-18599898@) (#2509)

* Upgrade Breadcrumb component

* Add test page

* Remove breadcrumb test page

kevinxh committed:
fix lint

kevinxh committed:
revert

bendvc committed:
Skip test because of Link component issue.

bendvc committed:
Merge branch 'feature/chakra-ui-upgrade-v3' into bendvc/W-18687117_nested_accordion_upgrade

bendvc committed:
Merge pull request #2512 from SalesforceCommerceCloud/bendvc/W-18687117_nested_accordion_upgrade

[Chakra V3 Upgrade 🎨] Update NestedAccordion Component @@W-18687117@@

bendvc committed:
Remove test page

bendvc committed:
Initial Commit

bendvc committed:
Update css for swatchGroup

adamraya committed:
[Chakra V3 Upgrade 🎨] Update SwatchGroup component (@W-18708628@) (#2515)

* Upgrade SwatchGroup component

* lint

* Remove test page from setup-app

kevinxh committed:
merge

adamraya committed:
[Chakra V3 Upgrade 🎨] Update Pagination component (@W-18700879@) (#2511)

* Upgrade pagination

* lint

* Clean up

* Remove test page

kevinxh committed:
Merge pull request #2505 from SalesforceCommerceCloud/kevinxh/W-18685099-checkout-skeleton

[Chakra V3 Upgrade 🎨] Update checkout skeleton Component @@W-18685099@@

bendvc committed:
Clean up styles

bendvc committed:
Merge branch 'feature/chakra-ui-upgrade-v3' into bendvc/W-18394804_pdp_upgrade_2

bendvc committed:
Cleanup and small style changes.

bendvc committed:
Manual revert of add to cart modal hook

bendvc committed:
Update packages/extension-chakra-storefront/src/components/product-view/index.jsx

Co-authored-by: Adam Raya <adamraya@users.noreply.github.com>
Signed-off-by: Ben Chypak <bchypak@mobify.com>

bendvc committed:
Initial Commit

bendvc committed:
Clean up

bendvc committed:
Update packages/extension-chakra-storefront/src/hooks/use-add-to-cart-modal.js

Co-authored-by: Adam Raya <adamraya@users.noreply.github.com>
Signed-off-by: Ben Chypak <bchypak@mobify.com>

bendvc committed:
Update packages/extension-chakra-storefront/src/hooks/use-add-to-cart-modal.js

Co-authored-by: Adam Raya <adamraya@users.noreply.github.com>
Signed-off-by: Ben Chypak <bchypak@mobify.com>

bendvc committed:
Update packages/extension-chakra-storefront/src/hooks/use-add-to-cart-modal.js

Co-authored-by: Adam Raya <adamraya@users.noreply.github.com>
Signed-off-by: Ben Chypak <bchypak@mobify.com>

bendvc committed:
Lint

bendvc committed:
Merge pull request #2523 from SalesforceCommerceCloud/bendvc/W-18718985_use_add_to_cart_modal_upgrade

[Chakra V3 Upgrade 🎨] Update useAddToCartModal Component @@W-18718985@@

bendvc committed:
Remove Testing Code

bendvc committed:
Merge branch 'feature/chakra-ui-upgrade-v3' into bendvc/W-18394804_pdp_upgrade_2

bendvc committed:
Fix tests

bendvc committed:
PR feedback

bendvc committed:
Re-silence tests

bendvc committed:
Merge pull request #2517 from SalesforceCommerceCloud/bendvc/W-18394804_pdp_upgrade_2

[Chakra V3 Upgrade 🎨] Update Product Detail Page @@W-18394804@@

bendvc committed:
Initial Commit

alexvuong committed:
W-18672556 WIP Chakra v3 header search (#2507)

* Upgrade search input component to chakra v3

bendvc committed:
Migrate the LocaleSelector component.

kevinxh committed:
migrate list menu themes

kevinxh committed:
migrate list menu popover

kevinxh committed:
migrate list menu popover trigger

kevinxh committed:
remove ChevronIconTrigger

bendvc committed:
Cleaning up styles

bendvc committed:
Clean up and fix tests

bendvc committed:
Merge branch 'feature/chakra-ui-upgrade-v3' into bendvc/W-18672605_drawer_menu_upgrade

kevinxh committed:
restructure listmenu

bendvc committed:
Cleaning up after AI

bendvc committed:
Re-add code AI removed

bendvc committed:
Fix tests for locale selector

adamraya committed:
[Chakra V3 Upgrade 🎨] Update Product List Page (@W-18599898@) (#2492)

* WIP Initial PLP Chakra v3 Upgrade

* WIP PLP basic layout working

* Migrate ProductTile

* Clean up

* Fix Wishlist icon

* Migrate Breadcrumb component

* Add styles with useSlotRecipe to breadcrumb component

* Undo CategoryLinks changes will do in separated Refinements PR

* Use css prop in ChakraBreadcrumb.Link

* Undo changes to DrawerMenu will be done in a separated PR

* Undo changes Header and ImageGallery

* Solve merge conflicts

* Undo NestedAccordion changes

* Move SizeRefinements to separate PR

* Move Breadcrumb changes to separate PR

* Fix SwatchGroup

* Enable pagination tests

* Fix pagination test name

* lint

* Enable ProductList component in routes

* Use Fade component

* Revert LocaleSelector changes

* Polish pagination styles to match

* Replace Sort Box/select with NativeSelect component

* lint

* Fix OfflineBanner change ...rest to ...props

* Simplify PLP ColorRefinements changes

bendvc committed:
Update packages/extension-chakra-storefront/src/components/drawer-menu/drawer-menu.jsx

Co-authored-by: Adam Raya <adamraya@users.noreply.github.com>
Signed-off-by: Ben Chypak <bchypak@mobify.com>

bendvc committed:
Merge branch 'feature/chakra-ui-upgrade-v3' into bendvc/W-18672605_drawer_menu_upgrade

bendvc committed:
PR feedback

bendvc committed:
Remove sample code

bendvc committed:
Merge branch 'bendvc/W-18672605_drawer_menu_upgrade' of https://github.com/SalesforceCommerceCloud/pwa-kit into bendvc/W-18672605_drawer_menu_upgrade

bendvc committed:
Merge pull request #2548 from SalesforceCommerceCloud/bendvc/W-18672605_drawer_menu_upgrade

[Chakra V3 Upgrade 🎨] Update drawerMenu Component @@W-18672605@@

bendvc committed:
Scaffold development page.

bendvc committed:
Migrate fields/forms; update styling

bendvc committed:
Fix link variant button styling

kevinxh committed:
add styles

kevinxh committed:
remove header text

kevinxh committed:
fix offset

kevinxh committed:
minor fixes

kevinxh committed:
lint

kevinxh committed:
add comments

kevinxh committed:
update jest config to add list menu test

kevinxh committed:
fix test file extension

kevinxh committed:
merge

kevinxh committed:
fix link hover

kevinxh committed:
fix hover

kevinxh committed:
lint

kevinxh committed:
lint

kevinxh committed:
remove unused import

kevinxh committed:
address pr feedbacks

kevinxh committed:
remove active link style

kevinxh committed:
Merge pull request #2549 from SalesforceCommerceCloud/W-18507835-chakra-v3-listmenu

[Chakra V3 Upgrade 🎨] Update listMenu Component @@W-18507835@@

vmarta committed:
[Chakra v3 upgrade] Migrating the AuthModal component (@W-18672998@) (#2555)

* 1st attempt at migrating AuthModal component

* Add temporary button to open the modal

* Migrating the Register form

* Update some theme styles

* Fix the recent merge conflicts

* Migrate the Reset Password form

* Migrate the content for passwordless flow

* Tweak the Sign In link

* Using the new asChild prop

* Remove the temp button on homepage

* Make sure useAuthModal hook return the same object shape

* Bring back the Account icon in the header

* Rename our Field component

* Remove unnecessary nested Link component

bendvc committed:
Merge branch 'feature/chakra-ui-upgrade-v3' into bendvc/W-18672820_login-form-upgrade

bendvc committed:
Merge latest

bendvc committed:
Remove test page

bendvc committed:
Clean up the buttons/link on the form

bendvc committed:
Revert changes in favour of those in vmarta's pr

kevinxh committed:
refactor pdp metadata

kevinxh committed:
add unit test

kevinxh committed:
add keywords

kevinxh committed:
re-use seo component

bendvc committed:
Merge pull request #2570 from SalesforceCommerceCloud/bendvc/W-18672820_login-form-upgrade

[Chakra V3 Upgrade 🎨] Update loginForm Component @@W-18672820@@

vmarta committed:
[Chakra v3 upgrade] Migrate the Login, Redirect, and Registration pages (@W-18672771@) (#2606)

* Re-enable some pages

* Migrate the social-login-redirect page

* Bring back the AlertIcon

* Bring back the AlertIcon

* Further migration the LoginForm component

* Add tests to run

kevinxh committed:
add plp metadata

kevinxh committed:
update implementation

kevinxh committed:
enable tests

kevinxh committed:
Merge branch 'feature/chakra-ui-upgrade-v3' into feature/refactor-page-metadata

Signed-off-by: Kevin He <kevin.he@salesforce.com>

vmarta committed:
Easier to add the test glob (#2618)

kevinxh committed:
lint

kevinxh committed:
Merge branch 'feature/refactor-page-metadata' of https://github.com/SalesforceCommerceCloud/pwa-kit into feature/refactor-page-metadata

kevinxh committed:
fix search description

kevinxh committed:
fix jest config

adamraya committed:
[Chakra V3 Upgrade 🎨] Update PLP Refinements components (@W-18394802@) (#2607)

* WIP Initial PLP refinements upgrade

* Migrate color-refinements

* Polish styles

* More polishing styles

* PR Feedback

* lint

* Update RadioRefinements to use onValueChange

* Add cursor pointer

* Fix styles SelectedRefinements background color

* Re-add clear filter and view items buttons of mobile refinements

* Use color black on mobile refinements buttons

* Add PLP tests to jest config

sf-henry-semaganda committed:
prepping chakra v3 upgrade

kevinxh committed:
fix test

kevinxh committed:
remove unused var

alexvuong committed:
W-18672731 Chakra v3 cart page (#2603)

* Chakra v3 migration cart page

kevinxh committed:
Merge branch 'feature/chakra-ui-upgrade-v3' into feature/refactor-page-metadata

vmarta committed:
[Chakra v3 upgrade] Migrate the ContactInfo section of the Checkout page (@W-18685076@ and @W-18685115@) (#2634)

* Checkout page now renders with some commented-out components

* 1st attempt at migrating ContactInfo

* Further migration

* Fix layerStyles

* Futher migration

* No need for icon's margin in this case

Since the wrapper Button already handles the whitespace for you.

* Use correct Button's variant

* Un-comment some code blocks

* Add missing props

* Style Button with colorPalette=red

* Fix linting

* Enable more tests

* Fix some merge conflicts

* Disable component for now because out of scope

* Use our own AlertIcon

sf-henry-semaganda committed:
fix linting

alexvuong committed:
set up page not found

alexvuong committed:
Merge remote-tracking branch 'origin/W-18672776_page_not_found' into W-18672776_page_not_found

alexvuong committed:
set up page not found

sf-henry-semaganda committed:
Merge branch 'feature/chakra-ui-upgrade-v3' into W-18672776_page_not_found

Signed-off-by: sf-henry-semaganda <henry.semaganda@salesforce.com>

kevinxh committed:
Merge pull request #2587 from SalesforceCommerceCloud/feature/refactor-page-metadata

@W-18820656 Refactor page level metadata

kevinxh committed:
move product details component

kevinxh committed:
fix bug where cause product bundle to not work

kevinxh committed:
create simple product and composite product

kevinxh committed:
rename partial pdp components

kevinxh committed:
create useProductDetailData hook

kevinxh committed:
fix copyright

kevinxh committed:
reorder imports

kevinxh committed:
extract analytics

kevinxh committed:
extract pdp wishlist logic

kevinxh committed:
extract recommended products section

kevinxh committed:
lint

kevinxh committed:
rename page metadata

kevinxh committed:
extract page cache

kevinxh committed:
add comments

kevinxh committed:
extract page-analytics

kevinxh committed:
lint

sf-henry-semaganda committed:
Merge pull request #2639 from SalesforceCommerceCloud/W-18672776_page_not_found

@W-18672776 feat: PageNotFound chakra v3 upgrade

sf-henry-semaganda committed:
remove changes on the default.json and setup-app.tsx

sf-henry-semaganda committed:
linting

sf-henry-semaganda committed:
Merge pull request #2676 from SalesforceCommerceCloud/revert_file_changes

@W-18672776 Revert file changes

kevinxh committed:
debug analytics

kevinxh committed:
remove console logs

kevinxh committed:
fix product set add to cart issue

kevinxh committed:
lint

kevinxh committed:
Merge pull request #2668 from SalesforceCommerceCloud/@W-18820722-refactor-pdp

@W-18820722 Refactor PDP for better readability and maintainability

unandyala committed:
Migrate Payment and PaymentForm to Chakra V3 (#2661)

Migrate Payment and PaymentForm to Chakra V3 (#2661)

sf-jie-dai committed:
[Chakra v3 upgrade] @W-18672580 update: migrate offline components from v2 to v3 (#2684)

* update: migrate offline components from v2 to v3

* add unit test coverage report & cleanup

* class name clean up

* use renderWithRouter to test offline-boundary

* use newer chakra UI component instead of basic html tags

* wrap user.click with act

* clean up comments

bendvc committed:
Migrate spinner and page-action-placeholder components

bendvc committed:
Add test pages for components.

bendvc committed:
Lint and remove test code

bendvc committed:
Merge pull request #2713 from SalesforceCommerceCloud/bendvc/W-18672828_migrate-loading-spinner

[ChakraV3 Upgrade] ♻️ Migrate LoadingSpinner and PageActionPlaceholder to Chakra UI

vcua-mobify committed:
@W-18685092@ @W-18685121@ Chakra v3 - update checkout shipping address and shipping option selection (#2690)

* Apply initial changes

* Radio card changes

* More radio card adjustments

* Fix shipping option selection

* Use value instead of selectedAddress for isSelected

* Lint

* Remove extra comment

---------

Co-authored-by: unandyala <unandyala@salesforce.com>

sf-henry-semaganda committed:
recommended pattern to define variants in the theme

sf-henry-semaganda committed:
add back explicit box sizing

sf-henry-semaganda committed:
remove unused import for v3, nt necessary

kzheng-sfdc committed:
@W-18910688 - break down cart page

sf-henry-semaganda committed:
remove redundant font size as its poart of theme

sf-henry-semaganda committed:
update

sf-henry-semaganda committed:
upgrade import as well

kzheng-sfdc committed:
lint fix

sf-henry-semaganda committed:
update

sf-henry-semaganda committed:
update linting

sf-henry-semaganda committed:
more uplifts

alexvuong committed:
W-18672799 Chakra UI v3 header migration (#2691)

* Migrate Header to Chakra v3
---------

Co-authored-by: Vincent Marta <vmarta@salesforce.com>

sf-henry-semaganda committed:
Merge branch 'feature/chakra-ui-upgrade-v3' into W-18397004

sf-henry-semaganda committed:
another pass at docs hence update

vcua-mobify committed:
@W-18672733@ Chakra v3 - Checkout Page follow up (#2723)

* Remove test data and enable disabled components on page

* Blue radiomark for shipping option selection

* Blue checkmark for billing address

* Fix icons

sf-henry-semaganda committed:
review fb update

sf-henry-semaganda committed:
updated theme to aply chakra v3 style

sf-henry-semaganda committed:
update revert default styling as should not apply to other pages

sf-henry-semaganda committed:
base styling instead of `default variant.

sf-henry-semaganda committed:
Merge branch 'feature/chakra-ui-upgrade-v3' into W-18397004

kzheng-sfdc committed:
tests for new hooks

kzheng-sfdc committed:
rename hook files

kzheng-sfdc committed:
renmae cartUtils to cart-utils

kzheng-sfdc committed:
lint

sf-henry-semaganda committed:
Merge pull request #2719 from SalesforceCommerceCloud/W-18397004

@W-18397004 feat: Refactor account addresses for chakra-v3

kzheng-sfdc committed:
added missing file

sf-jie-dai committed:
[Chakra V3 Upgrade]@W-18672570 migrate wishlist button/account page (#2727)

* enable wishlist button on header

* clean up comment

* unskip wishlist unit test

vcua-mobify committed:
@W-18685112@ Chakra V3 - credit card radio group (#2735)

* Migrate cc radio group

* Migrate from FormErrorMessage

* Adjust plus icon

* Apply feedback

* Lint

---------

Signed-off-by: vcua-mobify <47404250+vcua-mobify@users.noreply.github.com>

kevinxh committed:
extract PLP wishlist logic

kevinxh committed:
remove unused vars

kevinxh committed:
rename file for consistency

kevinxh committed:
add copyright

kevinxh committed:
add plp page cache

kevinxh committed:
remove dup code

kzheng-sfdc committed:
more tests

kzheng-sfdc committed:
renmaed use-cart-shipping to use-cart-default-shipping

unandyala committed:
Fix billing address (#2718)

* Fix billing address

kzheng-sfdc committed:
Merge pull request #2721 from SalesforceCommerceCloud/u/kzheng/W-18910688/break-down-cart-page

@W-18910688 - break down cart page

kevinxh committed:
extract sort component

kevinxh committed:
extract plp header

kevinxh committed:
rename page-header to title

kevinxh committed:
rename plp banner

kevinxh committed:
rename banner component

vcua-mobify committed:
@W-18672733@ Chakra v3 - Checkout page alert follow up (#2756)

* Fix alert

* Adjust alert

* Lint

bendvc committed:
Migrate to chakra v3

kzheng-sfdc committed:
@W-18910688 refactored cart-utils to error-utils and consolidate usage

kzheng-sfdc committed:
moved to hooks

alexvuong committed:
W-18672755 - W-18397045 - Chakra UI order history and order detail (#2730)

* Order history and order detail

kzheng-sfdc committed:
fixed linter

alexvuong committed:
W-18672586 - Update Error Component with Chakra v3 (#2776)

* Update Error component with chakra v3

kevinxh committed:
remove unused plp props

kevinxh committed:
remove plp above header component

kevinxh committed:
remove unused vars

kevinxh committed:
extract product grid and pagination

kevinxh committed:
add copyright

kzheng-sfdc committed:
Merge pull request #2781 from SalesforceCommerceCloud/u/kzheng/W-18910688/refactored-error-utils

@W-18910688 - refactored error utils

kevinxh committed:
rename files

kzheng-sfdc committed:
@W-18910688 fixed jest warnings

kzheng-sfdc committed:
Merge pull request #2791 from SalesforceCommerceCloud/u/kzheng/W-18910688/fixed-jest-warnings

@W-18910688 fixed jest warnings

sf-jie-dai committed:
[Chakra V3 Upgrade] @W-18672916 refactor Wishlist page components (#2768)

* migrate wishlist secondary button group

* migrate wishlist primary action

* migrate useDisclosure hook

* comment out failing tests

* add back button hover style

* use prependHandlersToServer in jest-setup to simplify server setup

* wrap user action in act to avoid React warning

alexvuong committed:
[Chakra v3] Clean up and enable tests for hooks folders (#2783)

Clean up warning in the terminal when running tests and re-enable tests in hook folders

bendvc committed:
Update packages/extension-chakra-storefront/src/pages/checkout/confirmation.jsx

Co-authored-by: Alex Vuong <alex.vuong@salesforce.com>
Signed-off-by: Ben Chypak <bchypak@mobify.com>

bendvc committed:
Update packages/extension-chakra-storefront/src/pages/checkout/confirmation.jsx

Co-authored-by: Alex Vuong <alex.vuong@salesforce.com>
Signed-off-by: Ben Chypak <bchypak@mobify.com>

bendvc committed:
Replace spacing with gap

bendvc committed:
Merge branch 'feature/chakra-ui-upgrade-v3' into bendvc/W-18913525_checkout-confirmation-page

bendvc committed:
Update jest.config.js

bendvc committed:
Update jest.config.js

unandyala committed:
@W-18672751: Migrate Account Details Page to Chakra V3 (#2789)

 Migrate Account Details Page to use Chakra V3

bendvc committed:
Merge pull request #2774 from SalesforceCommerceCloud/bendvc/W-18913525_checkout-confirmation-page

[ChakraV3 Upgrade] ♻️ Checkout Confirmation Page @@W-18913525@@

alexvuong committed:
W-19009476 Chakra UI v3 page designer comps (#2806)

* upgrade page designer comps with chakra v3

sf-jie-dai committed:
[Chakra V3 Upgrade] @W-18672757 refactor Wishlist page (#2801)

* migrate and enable wishlist page

* [Chakra V3 Upgrade] @W-18672916 refactor Wishlist page components (#2768)

* migrate wishlist secondary button group

* migrate wishlist primary action

* migrate useDisclosure hook

* comment out failing tests

* add back button hover style

* use prependHandlersToServer in jest-setup to simplify server setup

* wrap user action in act to avoid React warning

* skip commented tests to track them

* fix confirmation modal header style

* fix loading state for useWishlist hook

* center the text for empty wishlist/order history pages

* simplify code with Flex

* use string number for gap

* fix perttier issue

* Update packages/extension-chakra-storefront/src/pages/account/wishlist/index.jsx

Co-authored-by: Alex Vuong <alex.vuong@salesforce.com>
Signed-off-by: sf-jie-dai <jie.dai@salesforce.com>

* fix perttier issue

---------

Signed-off-by: sf-jie-dai <jie.dai@salesforce.com>
Co-authored-by: Alex Vuong <alex.vuong@salesforce.com>

alexvuong committed:
Replace Divider with Separator (#2809)

alexvuong committed:
W-19001098 Port over pwless bug fix in v4 (#2810)

* Fix pwless race conditions for all login places

vmarta committed:
[Chakra v3] data fetching during SSR rendering now works again (@W-18822712@) (#2785)

* Remove call to withLegacyGetProps

* Debugging

* Revert change to AppConfig

* Add todos

* Try commenting out the problematic Chakra hooks

Including those components that also call `useDisclosure`.

* Revert "Try commenting out the problematic Chakra hooks"

This reverts commit 970b97e6f487920b9c21282f4126733efca7ed23.

* Create a safe version of useDisclosure

* Fix react-ssr-prepass lack of no-op for useInsertionEffect

* Don't render Portal on server side

* Show header again

* Revert "Create a safe version of useDisclosure"

This reverts commit 669f2eb1941094e415db4936479afc10fd8970ae.

* This Portal is already rendered only on client side

* Render nothing on server side

* Clean up code

* Remove unnecessary category check

* Add comment

* Update changelog files

* Existing tests are now passing again

* Revert change to the skeleton

* Add a comment closer to the prepass call

The setting of useInsertionEffect to a no-op had to be done in another file.

* Create a safe version of Portal

* Update changelog files

* Add comments to clarify

* Check first whether useInsertionEffect is defined or not

* Add more tests to pass test coverage

I didn't want to add test to react-rendering.test.js because my code change isn't much for testing. So I decided to add tests for another file instead.

* Add comment

alexvuong committed:
clean up test warnings (#2812)

adamraya committed:
[Chakra V3 Upgrade 🎨] Create Skip Nav Components (@W-18507873@) (#2757)

* init

* Clean up styles

* PR Feedback

* Use zIndex value

* Use zIndex default skipNav value

* PR Feedback

* Fix copyright year

* Use token values instead of px

* Move styles to base

alexvuong committed:
Cleanup/chakra UI v3 part 2 (#2821)

* clean up and enable tests for all files

kevinxh committed:
extract plp data hook

kevinxh committed:
refine wishlist action

kevinxh committed:
extract page analytics

kevinxh committed:
lint

kevinxh committed:
add product-list-grid test

kevinxh committed:
add product-list-header test

kevinxh committed:
add product-list-title test

kevinxh committed:
add product-list-pagination test

kevinxh committed:
extract the plp control hook

kevinxh committed:
lint

kevinxh committed:
add use-product-list-controls.test.js

kevinxh committed:
add use-product-list-data unit test

kevinxh committed:
lint

sf-jie-dai committed:
fix event bubbling under product tile wishlist button (#2831)

kevinxh committed:
Merge pull request #2826 from SalesforceCommerceCloud/refactor-plp

@W-18910676 Refactor product listing page
